connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1870,7 +1880,16 @@ private Callable<Void> getConnectorStoppingCallable(final String connectorName)
         };
     }
 
-    private void reconfigureConnectorTasksWithRetry(long initialRequestTime, final String connName) {
+    /**
+     * Request task configs from the connector and write them to the config storage in case the configs are detected to
+     * have changed. This method retries infinitely in case of any errors.
```

Comment:
I am curious if there is a way to not do infinite retries. If we are actually retrying infinitely, esp in the case of `startConnector` phase, then the connector just doesn't have tasks. Is it possible to somehow bubble up errors as part of connector (not task) status?

Reply:
Hm, that's an interesting idea and I don't see the harm in limiting the number of retries to some reasonable value and then marking the connector as failed after that (we could include the last exception's trace in the connector's status).

> Is it possible to somehow bubble up errors as part of connector (not task) status?

The `AbstractHerder` (`DistributedHerder`'s parent class) implements the `ConnectorStatus.Listener` interface and so we should be able to update the connector's status to failed via https://github.com/apache/kafka/blob/81b3b2fb3399ab2784eb8158564fc1c9a1299a8d/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java#L191-L195

For instance, [here](https://github.com/apache/kafka/blob/81b3b2fb3399ab2784eb8158564fc1c9a1299a8d/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L1856) we use the `onFailure` hook to update a connector's status as failed if there is an exception thrown during startup.